programs/svm-spoke/src/lib.rs

```diff
@@ -104,11 +137,113 @@ pub mod svm_spoke {
+    ///   incrementing Id, stored in state. Seed: ["root_bundle",state.seed,root_bundle_id]
+    /// - system_program (Program): The system program required for account creation.
+    ///
+    /// ### Parameters:
+    /// - relayer_refund_root: Merkle root of the relayer refund tree.
+    /// - slow_relay_root: Merkle root of the slow relay tree.
+    pub fn relay_root_bundle(
+        ctx: Context<RelayRootBundle>,
+        relayer_refund_root: [u8; 32],
+        slow_relay_root: [u8; 32],
+    ) -> Result<()> {
+        instructions::relay_root_bundle(ctx, relayer_refund_root, slow_relay_root)
+    }
+
+    /// Deletes a root bundle in case of emergencies where a bad bundle has reached the Spoke. Only callable by owner.
+    ///
+    /// Will close the PDA for the associated root bundle_id. If used does not decrement state.root_bundle_id.
+    ///
+    /// ### Accounts:
+    /// - signer (Signer): The account that must be the owner to authorize the deletion.
+    /// - closer (SystemAccount): The account that will receive the lamports from closing the root_bundle account.
+    /// - state (Writable): Spoke state PDA. Seed: ["state",seed] where seed is 0 on mainnet.
+    /// - root_bundle (Writable): The root bundle PDA to be closed. Seed: ["root_bundle", state.seed, root_bundle_id].
+    ///
+    /// ### Parameters:
+    /// - root_bundle_id: Index of root bundle that needs to be deleted.
+    pub fn emergency_delete_root_bundle(
+        ctx: Context<EmergencyDeleteRootBundleState>,
+        root_bundle_id: u32,
+    ) -> Result<()> {
+        instructions::emergency_delete_root_bundle(ctx, root_bundle_id)
+    }
+
+    /****************************************
+     *          DEPOSIT FUNCTIONS           *
+     ****************************************/
+
+    /// Request to bridge input_token to a target chain and receive output_token.
+    ///
+    /// The fee paid to relayers and the system should be is captured in the spread between input and output amount
+    /// when adjusted to be denominated in the input token. A relayer on the destination chain will send output_amount
+    /// of output_token to the recipient and receive input_tokenon a repayment chain of their choice. Therefore,
+    /// the fee should account for destination fee transaction costs, the relayer's opportunity cost of capital while
+    /// they wait to be refunded following an optimistic challenge window in the HubPool, and the system fee that
+    /// they'll be charged. On the destination chain, the hash of the deposit data will be used to uniquely identify
+    /// this deposit, so modifying any params in it will result in a different hash and a different deposit. The hash
+    /// will comprise all parameters to this function along with this chain's chainId(). Relayers are only refunded for
+    /// filling deposits with deposit hashes that map exactly to the one emitted by this contract.
```

Comment:
Improve readability and fix some typos
```suggestion
    /// The fee paid to relayers and the system is captured in the spread between the input and output amounts,
    /// denominated in the input token. A relayer on the destination chain will send `output_amount` of `output_token`
    /// to the recipient and receive `input_token` on a repayment chain of their choice.
    ///
    /// The fee accounts for:
    /// - Destination transaction costs,
    /// - The relayer's opportunity cost of capital while waiting for a refund during the optimistic challenge window in the HubPool,
    /// - The system fee charged to the relayer.
    ///
    /// On the destination chain, a unique hash of the deposit data is used to identify this deposit. Modifying any
    /// parameters will result in a different hash, creating a separate deposit. The hash is computed using all parameters
    /// of this function along with the chain's `chainId()`. Relayers are refunded only for deposits with hashes that
    /// exactly match those emitted by this contract.
```